src/parameterizations/vertical/MOM_ALE_sponge.F90

```diff
@@ -1006,7 +1021,8 @@ subroutine apply_ALE_sponge(h, dt, G, GV, US, CS, Time)
 
       call pass_var(sp_val,G%Domain)
       do j=CS%jsc,CS%jec; do I=CS%iscB,CS%iecB
-       sp_val_u(I,j,1:nz_data) = 0.5*(sp_val(i,j,1:nz_data)+sp_val(i+1,j,1:nz_data))
+        sp_val_u(I,j,1:nz_data) = 0.5*(sp_val(i,j,1:nz_data)+sp_val(i+1,j,1:nz_data))
+        mask_u(I,j,1:nz_data) = max(mask_z(i,j,1:nz_data),mask_z(i+1,j,1:nz_data))
```

Comment:
Taking the average of the sp_val will potentially include a masked valued (huge value if set that way in the input file) and it will not be masked by taking the max of the mask_z. So I would suggest, if we want to keep the max of mask idea to weigh the sp_val by mask_z:
sp_val_u=(sp_val(i)*mask_z(i) + sp_val(i+1)*mask_z(i+1)) / max(mask_z(i)+mask_z(I+1),1e-10)

Reply:
@abozec This PR does not contain #1434 which redefines mask_uv and I believe resolves masking issues.   Specifically, the sponge_uv is turned off at OBC points and adjacent to land in #1434.  Sponge target masked values in the interior would cause a problem,  but that shouldn't happen.

Reply:
I assume the `mask_z` update is following the 0-or-1 convention as other MOM masks, so that `mask_u` is water (=1) if either east/west c-point is water.

I want to believe that a similar masking convention would have been applied to `sp_val` in `horiz_interp_and_extrap_tracer`, but I could be wrong here.

If there is some large value in `sp_val` caused by an input mask value, then perhaps this should be addressed in `horiz_interp_and_extrap_tracer`?

Reply:
I see from Matt's comment and #1434 that `max` was flipped to `min`, extending the mask.  I also do not see any special handling of input mask values in `horiz_interp_and_extrap_tracer`.

https://github.com/NOAA-GFDL/MOM6/blob/52f906cfc458ee77630f75c9d65da52a87fcfa70/src/framework/MOM_horizontal_regridding.F90#L891-L900

Maybe this is clobbering out any potentially large values causing by a different input mask value?

Reply:
That piece of code in MOM_horizontal_regridding does not filter all the land values for u and v, I'm not sure why... so a mask value in sponge seems to be the solution. 
I've tried the max and Matt's min method and both work in my example. It now depends on  if users prefer an extended mask (using min) or using the 1  value available (using max) to calculate the u and v sponge values.

Reply:
@abozec Is this something you need to be fixed before we merge?  Or can it be addressed in the next update to main?

Reply:
it can be addressed with #1434 of Matt's so I'm good with that.

Reply:
Ok great, we will merge 1434 in the next dev/gfdl -> main merge, which is expected to happen in a couple of weeks.